gtk4/src/subclass/cell_area_context.rs

```diff
@@ -0,0 +1,208 @@
+                    natural_size.as_mut_ptr(),
+                );
+                Some((minimum_size.assume_init(), natural_size.assume_init()))
+            } else {
+                None
+            }
+        }
+    }
+
+    fn parent_allocate(&self, cell_area_context: &Self::Type, width: i32, height: i32) {
+        unsafe {
+            let data = T::type_data();
+            let parent_class =
+                data.as_ref().get_parent_class() as *mut ffi::GtkCellAreaContextClass;
+            if let Some(f) = (*parent_class).allocate {
+                f(
+                    cell_area_context
+                        .unsafe_cast_ref::<CellAreaContext>()
+                        .to_glib_none()
+                        .0,
+                    width,
+                    height,
+                )
+            }
+        }
+    }
+}
+
+unsafe impl<T: CellAreaContextImpl> IsSubclassable<T> for CellAreaContext {
+    fn override_vfuncs(class: &mut glib::Class<Self>) {
+        <Object as IsSubclassable<T>>::override_vfuncs(class);
+
+        let klass = class.as_mut();
+        klass.reset = Some(cell_area_context_reset::<T>);
+        klass.get_preferred_height_for_width =
+            Some(cell_area_context_get_preferred_height_for_width::<T>);
+        klass.get_preferred_width_for_height =
+            Some(cell_area_context_get_preferred_width_for_height::<T>);
+        klass.allocate = Some(cell_area_context_allocate::<T>);
+    }
+}
+
+unsafe extern "C" fn cell_area_context_reset<T: CellAreaContextImpl>(
+    ptr: *mut ffi::GtkCellAreaContext,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<CellAreaContext> = from_glib_borrow(ptr);
+
+    imp.reset(wrap.unsafe_cast_ref())
+}
+
+unsafe extern "C" fn cell_area_context_get_preferred_height_for_width<T: CellAreaContextImpl>(
+    ptr: *mut ffi::GtkCellAreaContext,
+    width: i32,
+    minimum_height: *mut libc::c_int,
+    natural_height: *mut libc::c_int,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<CellAreaContext> = from_glib_borrow(ptr);
+
+    let ret = imp.get_preferred_height_for_width(wrap.unsafe_cast_ref(), width);
+    if let Some((min_height, nat_height)) = ret {
+        *minimum_height = min_height;
+        *natural_height = nat_height;
+    }
+}
+
+unsafe extern "C" fn cell_area_context_get_preferred_width_for_height<T: CellAreaContextImpl>(
+    ptr: *mut ffi::GtkCellAreaContext,
+    height: i32,
+    minimum_width: *mut libc::c_int,
+    natural_width: *mut libc::c_int,
+) {
+    let instance = &*(ptr as *mut T::Instance);
+    let imp = instance.get_impl();
+    let wrap: Borrowed<CellAreaContext> = from_glib_borrow(ptr);
+
+    let ret = imp.get_preferred_width_for_height(wrap.unsafe_cast_ref(), height);
+    if let Some((min_width, nat_width)) = ret {
+        *minimum_width = min_width;
+        *natural_width = nat_width;
```

Comment:
You should probably always set these to something instead of leaving them uninitialized